packages/theme-check-common/src/checks/deprecated-global-app-block-type/index.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  LiquidBranchNamed,
+  LiquidConditionalExpression,
+  LiquidTag,
+  NamedTags,
+  NodeTypes,
+} from '@shopify/liquid-html-parser';
+import { Severity, SourceCodeType, LiquidCheckDefinition } from '../../types';
+
+export const DeprecatedGlobalAppBlockType: LiquidCheckDefinition = {
+  meta: {
+    name: 'DeprecatedGlobalAppBlockType',
+    code: 'deprecated-global-app-block-type',
+    severity: Severity.ERROR,
+    type: SourceCodeType.LiquidHtml,
+    docs: {
+      description: 'Check for deprecated global app block type `@global`',
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks/deprecated-global-app-block-type',
+    },
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function report(position: { start: number; end: number }) {
+      context.report({
+        message: 'The global app block type `@global` is deprecated. Use `@app` instead.',
+        startIndex: position.start,
+        endIndex: position.end,
+      });
+    }
+    function checkMarkup(markup: string | LiquidConditionalExpression): boolean {
```

Comment:
Nit: Typically functions that return booleans should be fairly explicit about the boolean value they're returning. It helps to prefix the name with `has` or `is`. In this case, Id propose something like `hasGlobal`.

You currently have two functions prefixed with `check`: 
- checkMarkup: returns a boolean
- checkBranch: calls `report` and returns void

Using conventional naming patterns would help make your intent clearer.

Reply:
good call out, will keep that in mind for future functions